File: frontend-react/src/services/taskService.js
const API_URL = "http://localhost:8080/tasks";

// GET all tasks
export async function getTasks() {
  const response = await fetch(API_URL);
  return await response.json();
}

// CREATE task
export async function createTask(task) {
  const response = await fetch(API_URL, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(task),
  });

  return await response.json();
}

// DELETE task
export async function deleteTask(id) {
  await fetch(`${API_URL}/${id}`, {
    method: "DELETE",
  });
}

// UPDATE task
export async function updateTask(id, updatedTask) {
  const response = await fetch(`${API_URL}/${id}`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(updatedTask),
  });

  return await response.json();
}

export async function login(user) {
  const response = await fetch(
    "http://localhost:8080/users/login",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(user),
    }
  );

  return await response.json();
}

export async function getTasksByUser(userId) {
  const response = await fetch(
    `http://localhost:8080/tasks/user/${userId}`
  );

  return await response.json();
}

File: backend/src/main/java/service/TaskService.java
package com.example.demo.service;

import com.example.demo.entity.Task;
import com.example.demo.repository.TaskRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TaskService {

    private final TaskRepository taskRepository;

    public TaskService(TaskRepository taskRepository) {
        this.taskRepository = taskRepository;
    }

    // GET ALL TASKS
    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    // CREATE TASK
    public Task createTask(Task task) {

        System.out.println("Task title: " + task.getTitle());

        if (task.getUser() != null) {
            System.out.println("User ID: " + task.getUser().getId());
        } else {
            System.out.println("User is NULL");
        }

        return taskRepository.save(task);
    }

    // UPDATE TASK
    public Task updateTask(Long id, Task updatedTask) {

        Task task = taskRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Task not found"));

        task.setTitle(updatedTask.getTitle());
        task.setDeadline(updatedTask.getDeadline());
        task.setPriority(updatedTask.getPriority());
        task.setStatus(updatedTask.getStatus());
        task.setDescription(updatedTask.getDescription());

        return taskRepository.save(task);
    }

    // DELETE TASK
    public void deleteTask(Long id) {
        taskRepository.deleteById(id);
    }

    public List<Task> getTasksByUser(Long userId) {
        return taskRepository.findByUserId(userId);
    }
}